### PR TITLE
fix(bigquery-usage): fix dataset name for sharded table

### PR DESCRIPTION
**Pull request by MugdhaHardikar-GSLab**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by treff7es — CHANGES_REQUESTED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/usage/bigquery_usage.py [hunk 1]
```diff
@@ -272,20 +272,16 @@ def is_temporary_table(self, prefix: str) -> bool:
     def remove_extras(self, sharded_table_regex: str) -> "BigQueryTableRef":
         # Handle partitioned and sharded tables.
         table_name: Optional[str] = None
+        shortened_table_name = self.table
+        # if table name ends in _* or * then we strip it as that represents a query on a sharded table
+        shortened_table_name = shortened_table_name.removesuffix("_*")
```
[the remark this answers is not in the document]
Comment (MugdhaHardikar-GSLab) [reply metadata-ingestion/src/datahub/ingestion/source/usage/bigquery_usage.py]: Done

File: metadata-ingestion/tests/unit/test_bigquery_usage_source.py [hunk 2]
```diff
@@ -194,3 +194,9 @@ def test_bigquery_ref_extra_removal():
     assert new_table_ref.table == "foo"
     assert new_table_ref.project == table_ref.project
     assert new_table_ref.dataset == table_ref.dataset
+
+    table_ref = BigQueryTableRef("project-1234", "dataset-4567", "foo_2016*")
```
[the remark this answers is not in the document]
Comment (MugdhaHardikar-GSLab) [reply metadata-ingestion/tests/unit/test_bigquery_usage_source.py:198]: AS per discussion with @treff7es not taking this up in current PR.

**Review by shirshanka — APPROVED** [review]
LGTM

**Review by treff7es — COMMENTED** [review]
lgtm

**Pull request merged**